Implementation of missing save() function of AzureLocalCarts OrderLine.

davidmcatee-azure opened the pull request:
Resolves #43.

When the user changes the quantity of an order line on the cart page, [the `save()` method of the `orderLine` object is called](https://github.com/azurestandard/website/blob/67c1965190f2e8a62661a427d60d0b8fb97f539c/app/views/partials/cart_product_rows.html#L26). If it's a local cart, nothing happens because the method is not implemented on the `AzureLocalCarts` version of `OrderLine`. This implementation of the missing save method calculates the orderLine based on the quantity entered and recalculates the cart totals.

wking commented:
Looks good to me.  I removed some trailing whitespace and filled in
the commit message a bit with 3f875d9 → bf82fac, so let me know if
that looks good to you.

I think we might be able to move the $localStorage.orderLines.push
outside the promise.  See discussion of _getOrderLines in 2bc24438
(azure-providers.js: Add AzureLocalCarts factory, 2016-02-02, #34).
Does that sound right to you?

davidmcatee-azure commented:
I revised the message because it sounded like I added the call to `_newOrderLine`, but I didn't. I just took out the unnecessary additional API request.  I also moved `$localStorage.orderLines.push` outside the promise as you suggested.

wking commented on azure-providers.js:
```diff
@@ -1140,23 +1150,17 @@ var azureProvidersModule = angular
+            line['packaged-product'] = productCode;
+            line['quantity-ordered'] = quantityOrdered;
+            line = this._newOrderLine(line);
+            $localStorage.orderLines.push(line.orderLine);
```
Shouldn't this be pushing `line` instead of `line.orderLine`?

wking replied:
Ah, never mind, this is just pushing into storage, and we're still pushing `line` into `_this.orderLines`.
